Index: libdrm-2.4.96/include/drm/drm_fourcc.h
===================================================================
--- libdrm-2.4.96.orig/include/drm/drm_fourcc.h
+++ libdrm-2.4.96/include/drm/drm_fourcc.h
@@ -179,6 +179,7 @@ extern "C" {
 #define DRM_FORMAT_NV61		fourcc_code('N', 'V', '6', '1') /* 2x1 subsampled Cb:Cr plane */
 #define DRM_FORMAT_NV24		fourcc_code('N', 'V', '2', '4') /* non-subsampled Cr:Cb plane */
 #define DRM_FORMAT_NV42		fourcc_code('N', 'V', '4', '2') /* non-subsampled Cb:Cr plane */
+#define DRM_FORMAT_NV12_10		fourcc_code('N', 'A', '1', '2')
 
 /*
  * 3 plane YCbCr
