Index: libdrm-2.4.96/rockchip/rockchip_drm.c
===================================================================
--- libdrm-2.4.96.orig/rockchip/rockchip_drm.c
+++ libdrm-2.4.96/rockchip/rockchip_drm.c
@@ -129,6 +129,31 @@ fail:
 	return NULL;
 }
 
+struct rockchip_bo *rockchip_bo_from_handle(struct rockchip_device *dev,
+			uint32_t handle, uint32_t flags, uint32_t size)
+{
+	struct rockchip_bo *bo;
+
+	if (size == 0) {
+		fprintf(stderr, "invalid size.\n");
+		return NULL;
+	}
+
+	bo = calloc(1, sizeof(*bo));
+	if (!bo) {
+		fprintf(stderr, "failed to create bo[%s].\n",
+				strerror(errno));
+		return NULL;
+	}
+
+	bo->dev = dev;
+	bo->handle = handle;
+	bo->size = size;
+	bo->flags = flags;
+
+	return bo;
+}
+
 /*
  * Destroy a rockchip buffer object.
  *
Index: libdrm-2.4.96/rockchip/rockchip_drmif.h
===================================================================
--- libdrm-2.4.96.orig/rockchip/rockchip_drmif.h
+++ libdrm-2.4.96/rockchip/rockchip_drmif.h
@@ -71,5 +71,7 @@ struct rockchip_bo *rockchip_bo_from_nam
 			uint32_t name);
 int rockchip_bo_get_name(struct rockchip_bo *bo, uint32_t *name);
 uint32_t rockchip_bo_handle(struct rockchip_bo *bo);
+struct rockchip_bo *rockchip_bo_from_handle(struct rockchip_device *dev,
+			uint32_t handle, uint32_t flags, uint32_t size);
 void *rockchip_bo_map(struct rockchip_bo *bo);
 #endif /* ROCKCHIP_DRMIF_H_ */
