Index: libdrm-2.4.96/xf86drm.c
===================================================================
--- libdrm-2.4.96.orig/xf86drm.c
+++ libdrm-2.4.96/xf86drm.c
@@ -659,13 +659,9 @@ static int drmOpenByName(const char *nam
                     drmFreeVersion(version);
                     id = drmGetBusid(fd);
                     drmMsg("drmGetBusid returned '%s'\n", id ? id : "NULL");
-                    if (!id || !*id) {
-                        if (id)
-                            drmFreeBusid(id);
-                        return fd;
-                    } else {
+                    if (id)
                         drmFreeBusid(id);
-                    }
+                    return fd;
                 } else {
                     drmFreeVersion(version);
                 }
